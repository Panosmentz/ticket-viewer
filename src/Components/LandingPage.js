import React, { useState, useContext } from "react";
import { makeStyles } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import Container from "@material-ui/core/Container";
import TextField from "@material-ui/core/TextField";
import Avatar from "@material-ui/core/Avatar";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
import { Redirect } from "react-router-dom";
import { StateContext } from "../Context/StateContext";

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(20),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  avatar: {
    margin: theme.spacing(1),
    backgroundColor: "#066c7a",
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submitButton: {
    margin: theme.spacing(3, 0, 2),
    backgroundColor: "#066c7a",
    "&:hover": {
      backgroundColor: "#066c7a",
    },
  },
  textfield: {
    "& .MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline": {
      borderColor: "#066c7a",
    },
    "& .MuiInputLabel-outlined.Mui-focused": {
      color: "#066c7a",
    },
  },
}));

function LandingPage() {
  const classes = useStyles();
  const { getTickets, isAuthenticated } = useContext(StateContext); //getTickets makes the API call, isAuthenticated is true only when the API call responds with 200 OK(user must be authenticated)
  const [formData, setFormData] = useState({ email: "", password: "" }); //set the current state of email and password
  const { email, password } = formData; //get email and password from formData to be passed on the getTickets function

  const onChange = (e) =>
    setFormData({ ...formData, [e.target.name]: e.target.value }); //monitor state changes when keys are entered in the textfields

  const onSubmit = async (e) => {
    e.preventDefault();
    getTickets(email, password); //API call
  };
  if (isAuthenticated === true) {
    //true on action dispatch if API response status is 200
    return <Redirect to="/tickets" />;
  }

  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <Avatar className={classes.avatar}>
          <LockOutlinedIcon />
        </Avatar>
        <Typography component="h1" variant="h5">
          Authenticate
        </Typography>
        <form className={classes.form} onSubmit={onSubmit}>
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            id="email"
            label="Email Address"
            name="email"
            autoComplete="email"
            autoFocus
            value={email}
            onChange={(e) => onChange(e)}
            className={classes.textfield}
          />
          <TextField
            variant="outlined"
            margin="normal"
            required
            fullWidth
            name="password"
            label="Password"
            type="password"
            id="password"
            autoComplete="current-password"
            value={password}
            onChange={(e) => onChange(e)}
            className={classes.textfield}
          />
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="primary"
            className={classes.submitButton}
          >
            Authenticate
          </Button>
        </form>
      </div>
    </Container>
  );
}

export default LandingPage;
