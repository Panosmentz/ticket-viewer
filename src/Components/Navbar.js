import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import ViewWeekIcon from '@material-ui/icons/ViewWeek';
import {useHistory} from "react-router-dom";

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
  },
  title: {
    flexGrow: 1,
  },
});

 function Navbar() {
    let history = useHistory();
  const classes = useStyles();

  function handleOnClick(){
    history.push("/");
  }
  
  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
        <IconButton
            edge="start"
            onClick={handleOnClick}
            color="inherit"
            aria-label="menu"
            >
                <ViewWeekIcon />
            </IconButton>
          <Typography variant="h6" className={classes.title}>
            Ticket Viewer
          </Typography>
        </Toolbar>
      </AppBar>
    </div>
  );
}

export default Navbar;