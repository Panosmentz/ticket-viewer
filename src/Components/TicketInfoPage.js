import React, { useContext } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import Paper from "@material-ui/core/Paper";
import Divider from "@material-ui/core/Divider";
import TextField from "@material-ui/core/TextField";
import IconButton from "@material-ui/core/IconButton";
import Button from "@material-ui/core/Button";
import MenuItem from "@material-ui/core/MenuItem";
import InputAdornment from "@material-ui/core/InputAdornment";
import SendIcon from "@material-ui/icons/Send";
import SubjectIcon from "@material-ui/icons/Subject";
import AccountCircle from "@material-ui/icons/AccountCircle";
import AccessTimeIcon from "@material-ui/icons/AccessTime";
import EqualizerIcon from "@material-ui/icons/Equalizer";
import PriorityHighIcon from "@material-ui/icons/PriorityHigh";
import MergeTypeIcon from "@material-ui/icons/MergeType";
import ArrowBackIcon from "@material-ui/icons/ArrowBack";
import Skeleton from "@material-ui/lab/Skeleton";
import { useHistory, useParams } from "react-router-dom";
import { StateContext } from "../Context/StateContext";

const useStyles = makeStyles({
  root: {
    width: "100%",
    height: "100%",
  },
  container: {
    maxHeight: 440,
  },
  infoPaper: {
    padding: 30,
    textAlign: "center",
    height: "auto",
  },
  ticketPaper: {
    padding: 30,
    textAlign: "left",
    height: "auto",
  },
  typography: {
    fontSize: 18,
  },
  margin: {
    marginTop: 20,
  },
  button: {
    marginTop: 100,
    color: "white",
    backgroundColor: "#03363d",
    "&:hover": {
      backgroundColor: "#066c7a",
    },
  },
  skeletonPaper: {
    height: "auto",
  },
  skeletonText: {
    height: 100,
  },
});

function TicketInfoPage() {
  const classes = useStyles();
  let history = useHistory();
  const { id } = useParams();
  const { tickets } = useContext(StateContext);
  const tbc = "TBC";

  const SendButton = () => (
    <IconButton>
      <SendIcon />
    </IconButton>
  );

  function onClick() {
    history.push("/tickets");
  }

  if (Object.keys(tickets).length === 0) {
    return (
      <div className={classes.root}>
        <Grid
          container
          spacing={0}
          direction="row"
          justify="center"
          alignItems="center"
        >
          <Grid item xs={12} md={2}>
            <Paper className={classes.skeletonPaper}>
              <Skeleton variant="text" className={classes.skeletonText} />
              <Divider />
              <Skeleton variant="text" className={classes.skeletonText} />
              <Divider />
              <Skeleton variant="text" className={classes.skeletonText} />
            </Paper>
          </Grid>
          <Grid item xs={12} md={10}>
            <Paper className={classes.skeletonPaper}>
              <Skeleton variant="text" className={classes.skeletonText} />
              <Divider />
              <Skeleton variant="text" className={classes.skeletonText} />
              <Divider />
              <Skeleton variant="text" className={classes.skeletonText} />
            </Paper>
          </Grid>
          <Button
            variant="contained"
            className={classes.button}
            startIcon={<ArrowBackIcon />}
            onClick={onClick}
          >
            Back to my tickets
          </Button>
        </Grid>
      </div>
    );
  }
  return (
    <div className={classes.root}>
      <Grid
        container
        spacing={0}
        direction="row"
        justify="center"
        alignItems="center"
      >
        <Grid item xs={12} md={2}>
          <Paper className={classes.infoPaper}>
            <Typography variant="h6" color="primary" className={classes.margin}>
              <AccessTimeIcon /> Date created :{" "}
              {new Date(tickets[id - 1].created_at).toLocaleDateString("en-gb")}
            </Typography>
            <Divider />
            <TextField
              className={classes.margin}
              id="requester"
              label="Requester"
              value={tickets[id - 1].requester_id}
              InputProps={{
                readOnly: true,
                startAdornment: (
                  <InputAdornment position="start">
                    <AccountCircle />
                  </InputAdornment>
                ),
              }}
            />
            <TextField
              className={classes.margin}
              id="assignee"
              label="Select Assignee"
              value={tickets[id - 1].assignee_id}
              InputProps={{
                readOnly: true,
                startAdornment: (
                  <InputAdornment position="start">
                    <AccountCircle />
                  </InputAdornment>
                ),
              }}
            />
            <Typography variant="h6" color="primary" className={classes.margin}>
              <PriorityHighIcon /> Priority :{" "}
              {tickets[id - 1].priority ? (
                <TextField
                  id="select-priority"
                  select
                  value={tickets[id - 1].priority}
                  helperText="Change ticket priority"
                >
                  <MenuItem
                    key={tickets[id - 1].priority}
                    value={tickets[id - 1].priority}
                  >
                    {tickets[id - 1].priority}
                  </MenuItem>
                </TextField>
              ) : (
                <TextField
                  id="select-priority"
                  select
                  value={tbc}
                  helperText="Change ticket priority"
                >
                  <MenuItem key={tbc} value={tbc}>
                    {"TBC"}
                  </MenuItem>
                </TextField>
              )}
            </Typography>

            <Typography variant="h6" color="primary" className={classes.margin}>
              <EqualizerIcon /> Status :{" "}
              <TextField
                id="select-status"
                select
                value={tickets[id - 1].status}
                helperText="Change ticket status"
              >
                {tickets[id - 1].status ? (
                  <MenuItem
                    key={tickets[id - 1].status}
                    value={tickets[id - 1].status}
                  >
                    {tickets[id - 1].status}
                  </MenuItem>
                ) : (
                  <MenuItem key={tbc} value={tbc}>
                    {tbc}
                  </MenuItem>
                )}
              </TextField>
            </Typography>

            <Typography variant="h6" color="primary" className={classes.margin}>
              <MergeTypeIcon />
              Type :{" "}
              {tickets[id - 1].type ? (
                <TextField
                  id="select-type"
                  select
                  value={tickets[id - 1].type}
                  helperText="Change ticket priority"
                >
                  <MenuItem
                    key={tickets[id - 1].type}
                    value={tickets[id - 1].type}
                  >
                    {tickets[id - 1].type}
                  </MenuItem>
                </TextField>
              ) : (
                <TextField
                  id="select-type"
                  select
                  value={tbc}
                  helperText="Change ticket type"
                >
                  <MenuItem key={tbc} value={tbc}>
                    {"TBC"}
                  </MenuItem>
                </TextField>
              )}
            </Typography>
          </Paper>
        </Grid>
        <Grid item xs={12} md={10}>
          <Paper className={classes.ticketPaper}>
            <Typography variant="h4" color="primary" gutterBottom="true">
              {" "}
              <SubjectIcon /> {tickets[id - 1].subject}{" "}
            </Typography>
            <Typography
              variant="subtitle2"
              gutterBottom="true"
              color="textSecondary"
            >
              Via {tickets[id - 1].via.channel}{" "}
            </Typography>
            <Divider />
            <Typography
              className={classes.typography}
              variant="h6"
              color="textPrimary"
            >
              {tickets[id - 1].description}
            </Typography>

            <TextField
              id="standard-full-width"
              multiline={true}
              rows={5}
              style={{ margin: 8, marginTop: 50 }}
              fullWidth
              margin="normal"
              InputProps={{ endAdornment: <SendButton /> }}
              label="Send message"
            />
          </Paper>
        </Grid>
        <Button
          variant="contained"
          className={classes.button}
          startIcon={<ArrowBackIcon />}
          onClick={onClick}
        >
          Back to my tickets
        </Button>
      </Grid>
    </div>
  );
}

export default TicketInfoPage;
