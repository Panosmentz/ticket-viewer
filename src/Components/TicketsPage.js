import React, { useContext, useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TablePagination from "@material-ui/core/TablePagination";
import TableRow from "@material-ui/core/TableRow";
import { useHistory } from "react-router-dom";
import { StateContext } from "../Context/StateContext";

const useStyles = makeStyles({
  root: {
    width: "100%",
  },
  container: {
    maxHeight: 800,
  },
});

function TicketsPage() {
  const classes = useStyles();
  let history = useHistory();
  const { tickets } = useContext(StateContext); //tickets object is stored in React's Context API and can be used from all Components
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(25); //only display 25 tickets at each page

  const columns = [
    { id: "id", label: "ID" },
    { id: "subject", label: "Subject" },
    {
      id: "requester",
      label: "Requester",
    },
    {
      id: "requesterupdated",
      label: "Requester updated",
    },
    {
      id: "group",
      label: "Group",
    },
    {
      id: "assignee",
      label: "Assignee",
    },
  ];

  const rows = []; //define an array and populate it with data from the tickets Object
  for (let i = 0; i < tickets.length; i++) {
    rows.push({
      id: tickets[i].id,
      subject: tickets[i].subject,
      requester: tickets[i].requester_id,
      requesterupdated: new Date(tickets[i].created_at).toLocaleDateString(
        "en-gb"
      ),
      group: tickets[i].group_id,
      assignee: tickets[i].assignee_id,
    });
  }

  const handleChangePage = (event, newPage) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event) => {
    setRowsPerPage(+event.target.value);
    setPage(0);
  };

  const handleRowClick = (event) => {
    history.push(`/tickets/${event}`); //if the row is clicked, redirect to /tickets/:id
  };

  return (
    <Grid className={classes.root}>
      <TableContainer className={classes.container}>
        <Table stickyHeader aria-label="sticky table">
          <TableHead>
            <TableRow>
              {columns.map((column) => (
                <TableCell key={column.id}>{column.label}</TableCell>
              ))}
            </TableRow>
          </TableHead>
          <TableBody>
            {rows
              .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
              .map((row) => {
                return (
                  <TableRow
                    tabIndex={-1}
                    key={row.code}
                    onClick={() => handleRowClick(row.id)}
                    hover={true}
                  >
                    {columns.map((column) => {
                      const value = row[column.id];
                      return (
                        <TableCell key={column.id}>
                          {column.format ? column.format(value) : value}
                        </TableCell>
                      );
                    })}
                  </TableRow>
                );
              })}
          </TableBody>
        </Table>
      </TableContainer>
      <TablePagination
        rowsPerPageOptions={[25]}
        component="div"
        count={rows.length}
        rowsPerPage={rowsPerPage}
        page={page}
        onChangePage={handleChangePage}
        onChangeRowsPerPage={handleChangeRowsPerPage}
      />
    </Grid>
  );
}

export default TicketsPage;
