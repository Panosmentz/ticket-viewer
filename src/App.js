import './App.css';
import LandingPage from "./Components/LandingPage"
import TicketsPage from "./Components/TicketsPage"
import React from 'react'
import {BrowserRouter as Router, Switch, Route} from "react-router-dom";
import {StateProvider} from "./Context/StateContext";

function App() {
  return (
    <StateProvider>
    <div>
    <Router>
      <Switch>
        <Route  exact path="/" component={LandingPage} />
        <Route exact path="/tickets" component={TicketsPage} />
      </Switch>
    </Router>
    </div>
    </StateProvider>
  )
}



export default App;
