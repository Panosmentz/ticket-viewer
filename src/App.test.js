import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import Navbar from "./Components/Navbar";
import LandingPage from "./Components/LandingPage";
import TicketsPage from "./Components/TicketsPage";
import TicketInfoPage from "./Components/LandingPage";
import toJson from "enzyme-to-json";

it("Renders Navbar", () => {
  const wrapper = shallow(<App />);
  const navbar = wrapper.find(Navbar);
  expect(navbar.exists()).toBe(true);
});

//create snapshots of the App and saves them in the _snapshots_ folder, can monitor changes
describe("snapshots", () => {
  it("App snapshots", () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });
  it("Landing Page snapshots", () => {
    const landingPageTree = shallow(<LandingPage />);
    expect(toJson(landingPageTree)).toMatchSnapshot();
  });
  it("Navbar snapshots", () => {
    const navbarTree = shallow(<Navbar />);
    expect(toJson(navbarTree)).toMatchSnapshot();
  });
  it("Tickets Page snapshots", () => {
    const ticketsPageTree = shallow(<TicketsPage />);
    expect(toJson(ticketsPageTree)).toMatchSnapshot();
  });
  it("Ticket Info Page snapshots", () => {
    const ticketInfoPageTree = shallow(<TicketInfoPage />);
    expect(toJson(ticketInfoPageTree)).toMatchSnapshot();
  });
});
