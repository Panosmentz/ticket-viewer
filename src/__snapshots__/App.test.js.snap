// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots App snapshots 1`] = `
<StateProvider>
  <ToastContainer
    autoClose={5000}
    closeButton={[Function]}
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    role="alert"
    rtl={false}
    transition={[Function]}
  />
  <BrowserRouter>
    <Navbar />
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/tickets"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/tickets/:id"
      />
    </Switch>
  </BrowserRouter>
</StateProvider>
`;

exports[`snapshots Landing Page snapshots 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="xs"
>
  <WithStyles(CssBaseline) />
  <div
    className="makeStyles-paper-1"
  >
    <WithStyles(ForwardRef(Avatar))
      className="makeStyles-avatar-2"
    >
      <LockOutlinedIcon />
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(Typography))
      component="h1"
      variant="h5"
    >
      Authenticate
    </WithStyles(ForwardRef(Typography))>
    <form
      className="makeStyles-form-3"
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(TextField))
        autoComplete="email"
        autoFocus={true}
        className="makeStyles-textfield-5"
        fullWidth={true}
        id="email"
        label="Email Address"
        margin="normal"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
        variant="outlined"
      />
      <WithStyles(ForwardRef(TextField))
        autoComplete="current-password"
        className="makeStyles-textfield-5"
        fullWidth={true}
        id="password"
        label="Password"
        margin="normal"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
        variant="outlined"
      />
      <WithStyles(ForwardRef(Button))
        className="makeStyles-submitButton-4"
        color="primary"
        fullWidth={true}
        type="submit"
        variant="contained"
      >
        Authenticate
      </WithStyles(ForwardRef(Button))>
    </form>
  </div>
</WithStyles(ForwardRef(Container))>
`;

exports[`snapshots Navbar snapshots 1`] = `
<div
  className="makeStyles-root-6"
>
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-7"
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="menu"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <ViewWeekIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-8"
        variant="h6"
      >
        Ticket Viewer
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;

exports[`snapshots Ticket Info Page snapshots 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="xs"
>
  <WithStyles(CssBaseline) />
  <div
    className="makeStyles-paper-1"
  >
    <WithStyles(ForwardRef(Avatar))
      className="makeStyles-avatar-2"
    >
      <LockOutlinedIcon />
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(Typography))
      component="h1"
      variant="h5"
    >
      Authenticate
    </WithStyles(ForwardRef(Typography))>
    <form
      className="makeStyles-form-3"
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(TextField))
        autoComplete="email"
        autoFocus={true}
        className="makeStyles-textfield-5"
        fullWidth={true}
        id="email"
        label="Email Address"
        margin="normal"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
        variant="outlined"
      />
      <WithStyles(ForwardRef(TextField))
        autoComplete="current-password"
        className="makeStyles-textfield-5"
        fullWidth={true}
        id="password"
        label="Password"
        margin="normal"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
        variant="outlined"
      />
      <WithStyles(ForwardRef(Button))
        className="makeStyles-submitButton-4"
        color="primary"
        fullWidth={true}
        type="submit"
        variant="contained"
      >
        Authenticate
      </WithStyles(ForwardRef(Button))>
    </form>
  </div>
</WithStyles(ForwardRef(Container))>
`;

exports[`snapshots Tickets Page snapshots 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-root-9"
>
  <WithStyles(ForwardRef(TableContainer))
    className="makeStyles-container-10"
  >
    <WithStyles(ForwardRef(Table))
      aria-label="sticky table"
      stickyHeader={true}
    >
      <WithStyles(ForwardRef(TableHead))>
        <WithStyles(ForwardRef(TableRow))>
          <WithStyles(ForwardRef(TableCell))
            key="id"
          >
            ID
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))
            key="subject"
          >
            Subject
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))
            key="requester"
          >
            Requester
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))
            key="requesterupdated"
          >
            Requester updated
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))
            key="group"
          >
            Group
          </WithStyles(ForwardRef(TableCell))>
          <WithStyles(ForwardRef(TableCell))
            key="assignee"
          >
            Assignee
          </WithStyles(ForwardRef(TableCell))>
        </WithStyles(ForwardRef(TableRow))>
      </WithStyles(ForwardRef(TableHead))>
      <WithStyles(ForwardRef(TableBody)) />
    </WithStyles(ForwardRef(Table))>
  </WithStyles(ForwardRef(TableContainer))>
  <WithStyles(ForwardRef(TablePagination))
    component="div"
    count={0}
    onChangePage={[Function]}
    onChangeRowsPerPage={[Function]}
    page={0}
    rowsPerPage={25}
    rowsPerPageOptions={
      Array [
        25,
      ]
    }
  />
</WithStyles(ForwardRef(Grid))>
`;
